src/issue.py

```diff
@@ -0,0 +1,40 @@
+
+class Issue:
+
+    def __init__(self):
+        self.url = None  # str
+        self.token = None  # str
+        self.headers = None  # dict[str, str]
+
+        self.assignee = None  # str
+        self.created = None  # datetime object
+        self.description = None  # str
+        self.github_key = None  # str, this identifier is used by zenhub and github
+        self.github_repo_name = None  # str
+        self.issue_type = None  # str
+        self.jira_key = None  # str, this identifier is only used by jira
+        self.parent = None  # str, the epic that this issue is a sub-task of
+        self.status = None  # str
+        self.story_points = None  # int
+        self.summary = None  # str
+        self.updated = None  # datetime object
+
+    def update_from(self, source: 'Issue'):
+        """
+        Set all fields in the sink issue (self) to match those in the source Issue object.
+        Fields that are defined in self but are None in source will be left alone.
+        """
+
+        for attribute in source.__dict__.keys():
+            if attribute not in ['headers', 'url', 'token', 'description']:  # ignore attributes specific to the source
+                if source.__dict__[attribute] is not None:
```

Comment:
Replaced this with a list comprehension using update(). I'm not sure if this is possible for the method below because it checks the value of the dict being updated, not of the source